src/include/serial.i

```diff
@@ -312,11 +312,11 @@ __wt_update_serial(WT_SESSION_IMPL *session, WT_PAGE *page,
 	 * is used as an indicator of there being further updates on this page.
 	 */
 	if ((txn = page->modify->obsolete_check_txn) != WT_TXN_NONE) {
-		if (!__wt_txn_visible_all(session, txn)) {
+		if (!__wt_txn_visible_all(session, txn, NULL)) {
 			/* Try to move the oldest ID forward and re-check. */
 			WT_RET(__wt_txn_update_oldest(session, 0));
 
-			if (!__wt_txn_visible_all(session, txn))
+			if (!__wt_txn_visible_all(session, txn, NULL))
```

Comment:
This is just an optimization (don't retry obsolete checks if nothing has changed).  I don't think it needs to be fixed.